src/main/java/com/wavefront/sdk/common/application/ApplicationTags.java

```diff
@@ -82,7 +85,45 @@ public Builder shard(String shard) {
      * @return {@code this}
      */
     public Builder customTags(Map<String, String> customTags) {
-      this.customTags = customTags;
+      this.customTags.putAll(customTags);
+      return this;
+    }
+
+    /**
+     * Set additional custom tags from environment variables that matches the a
```

Comment:
`Set additional custom tags from environment variables that match the given regex prefix`.

Can remove the part about this being optional. It's implict in the builder pattern.